### PR TITLE
feat: support more protocol 0 devices

### PR DESCRIPTION
**Pull request by bhuebschen**
enhanced VentaDeviceType
added Filter-Time
added CleaningMode
added Humidity as sensor

### CODE REVIEW

File: custom_components/venta/sensor.py [hunk 1]
```diff
@@ -140,6 +169,16 @@ class VentaSensorEntityDescription(
         is not None,
         value_func=lambda coordinator: coordinator.data.measure.get("WaterLevel"),
     ),
+    VentaSensorEntityDescription(
```
Comment (bhuebschen) [inline custom_components/venta/sensor.py:172]: "Dust" is shown as PM2.5 in the Venta-App

File: custom_components/venta/binary_sensor.py [hunk 2]
```diff
@@ -100,10 +102,24 @@ def _supported_sensors(
...
+                VentaBinarySensorEntityDescription(
+                    key=ATTR_NEEDS_CLEANING,
+                    translation_key="needs_cleaning",
+                    icon="mdi:filter",
+                    value_func=lambda data: needs_maintenance(
+                        data.info.get("FilterT"), FILTER_TIME_DAYS
+                    ),
+                ),
+                VentaBinarySensorEntityDescription(
+                    key=ATTR_CLEANING,
```
[the remark this answers is not in the document]
Comment (bhuebschen) [reply custom_components/venta/binary_sensor.py:118]: it represents that the device is currently cleaning (and cannot be used while cleaning)

File: custom_components/venta/venta.py [hunk 3]
```diff
@@ -30,8 +30,15 @@ class VentaDeviceType(Enum):
...
-    AH902 = 12
+    LP60 = 1
+    LPH60 = 2
+    LW60 = 3,
+    LW60T = 4,
+    LW62 = 5,
+    LW62T = 6,
+    AP902 = 11,
+    AH902 = 12,
+    AW902 = 13,
```
[the remark this answers is not in the document]
Comment (bhuebschen) [reply custom_components/venta/venta.py:41]: i've tested it using the "Venta Home"-App on my iPhone and this part implemented in Node-Red:

```
function str2ab(str) {
    var buf = new ArrayBuffer(str.length); // 2 bytes for each char
    var bufView = new Uint8Array(buf);
    for (var i = 0, strLen = str.length; i < strLen; i++) {
        bufView[i] = str.charCodeAt(i);
    }
    return buf;
}

var str = '{"Header":{"DeviceType":1,"IpAdress":"10.10.0.9","MacAdress":"00:00:00:00:00:01"}}';

return {
    "payload": Buffer.from(str2ab("Content-Length: 82\r\n\r\n" + str))
}
``
<img width="748" alt="Bildschirmfoto 2024-04-02 um 13 43 10" src="https://github.com/Michsior14/ha-venta/assets/1864448/dfdaf29f-f321-4ee8-984d-a3d5c1689ba0">
`

File: custom_components/venta/sensor.py [hunk 4]
```diff
@@ -130,6 +149,16 @@ class VentaSensorEntityDescription(
         is not None,
         value_func=lambda coordinator: coordinator.data.measure.get("Temperature"),
     ),
+    VentaSensorEntityDescription(
```
[the remark this answers is not in the document]
Comment (bhuebschen) [reply custom_components/venta/sensor.py:152]: LPH60 WIFI APP CONTROL AIRWASHER

Hybrid – Premium Air Purifier with Humidification 
Room size – purifies up to 45m², humidifies up to 95 m² 
Coarse dust filter (washable) 
Allergen, virus and fine dust filter removes particulates down to a size of 0.1 μm 
Active carbon filter eliminates odours/gases down to particle sizes of 0.0002 μm (optional) 
Unique sensor technology – intelligent measurement of air quality and humidity for separately controlled air purification and humidification functions
Hygienic cold evaporation without humidifier mats 
Uses tap water 
Patented hygiene disk for pure water without chemical additives, low-maintenance continuous operation 

https://www.venta-air.com/en_de/detail/8d0ab9f571244f1a8675c0a757147344

File: custom_components/venta/sensor.py (hunk 1 above)
Comment (bhuebschen) [reply custom_components/venta/sensor.py:172]: 7988406B-C186-49D9-86CB-5E7C154405F5_1_102_o

Comment (bhuebschen) [reply custom_components/venta/sensor.py:172]:
```
{
   "Header":{
      "DeviceType":2,
      "MacAdress":"f8...:9c",
      "Hash":"...",
      "Error":0,
      "DeviceName":""
   },
   "Action":{
      "Power":true,
      "FanSpeed":2,
      "TargetHum":55,
      "Timer":0,
      "Boost":false,
      "SleepMode":false,
      "ChildLock":false,
      "Automatic":true,
      "SysLanguage":3,
      "CleanLanguage":1,
      "TempUnit":0,
      "DisplayLeft":0,
      "DisplayRight":1,
      "Reset":0,
      "ConINet":0,
      "DelUser":false
   },
   "Info":{
      "SWDisplay":"V2.009",
      "SWPower":"V1.001",
      "SWTouch":"V1.3.0",
      "SWWIFI":"V19.4.8",
      "CleanMode":false,
      "RelState":[
         false,
         false,
         false,
         false
      ],
      "TimerT":0,
      "OperationT":147640,
      "DiscIonT":26401,
      "CleaningT":52801,
      "FilterT":19596,
      "UVCOnT":0,
      "UVCOffT":0,
      "CleaningR":0,
      "Warnings":97
   },
   "Measure":{
      "Temperature":24.4,
      "Humidity":39.1,
      "Dust":1,
      "WaterLevel":2,
      "FanRpm":0
   }
}
```

File: custom_components/venta/translations/de.json [hunk 5]
```diff
@@ -5,15 +5,15 @@
         },
         "error": {
             "cannot_connect": "Verbindung fehlgeschlagen",
-            "cannot_detect_api_version": "Kann API-Version nicht bestimmen",
+            "cannot_detect_api_version": "Kann API-Version nicht ermitteln",
```
[the remark this answers is not in the document]
Comment (bhuebschen) [reply custom_components/venta/translations/de.json:8]: as you can see on lokalise.com.. i already made these changes

File: custom_components/venta/venta.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bhuebschen) [reply custom_components/venta/venta.py:41]: i tested from 1 up to 25, 100 to 110 and 145 to 155 ... and made the changes as the device types shown up in the app

[the remark this answers is not in the document]
Comment (bhuebschen) [reply custom_components/venta/venta.py:41]: searching in the venta-home-app for Android, i found this:

public enum DeviceModelId
{
    AH500 = 500,
    AH902 = 12,
    AP902 = 11,
    AS100 = 100,
    AS150 = 150,
    AW902 = 13,
    LP60 = 1,
    LP73 = 0x74,
    LP74 = 0x75,
    LPH60 = 2,
    LW60 = 3,
    LW60T = 4,
    LW62 = 5,
    LW62T = 6,
    LW73 = 0x6a,
    LW74 = 0x6b
}

[the remark this answers is not in the document]
Comment (Michsior14) [reply custom_components/venta/venta.py:41]: Ok, thanks for such a deep analysis ;)

**Comment by Michsior14** [comment]
Will have a look on this today/tomorrow evening ;)

**Review by Michsior14 — APPROVED** [review]
@bhuebschen Thanks a lot!

**Pull request merged**